File: fe/lib/models/flashcard.dart
class Flashcard {
  final int id;
  final int topicId;
  final String frontText;
  final String? backText;
  final String? example;
  final String? audioUrl;

  Flashcard({required this.id, required this.topicId, required this.frontText, this.backText, this.example, this.audioUrl});

  factory Flashcard.fromJson(Map<String, dynamic> json) {
    return Flashcard(
      id: json['id'],
      topicId: json['topic_id'],
      frontText: json['front_text'] ?? '',
      backText: json['back_text'],
      example: json['example'],
      audioUrl: json['audio_url'],
    );
  }
}

File: be/app/Models/Flashcard.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Flashcard extends Model
{
    protected $fillable = [
        'topic_id',
        'front_text',
        'back_text',
        'example',
        'audio_url',
    ];

    public function topic()
    {
        return $this->belongsTo(Topic::class);
    }
}
